walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -121,6 +123,69 @@ public void everyAttendeeIsConsidered() {
     Assert.assertEquals(expected, actual);
   }
 
+  @Test
+  public void everyAttendeeIsConsideredAndOptionalIgnored() {
+    // Have each person have different events and an optional attendee who has a meeting the
+    // whole day. Since he doesn't have any available spots, he will be ignored and 3 ranges
+    // will be returned.
+    //
+    // Events  :       |--A--|     |--B--|
+    // Optional: |------------C----------------|
+    // Day     : |-----------------------------|
+    // Options : |--1--|     |--2--|     |--3--|
+
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartDuration(TIME_0800AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartDuration(TIME_0900AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_B)),
+        new Event("Event 3", TimeRange.fromStartDuration((TimeRange.START_OF_DAY, DURATION_2_HOUR * 12),
+            Arrays.asList(PERSON_C)));
+
+    MeetingRequest request =
+        new MeetingRequest(Arrays.asList(PERSON_A, PERSON_B), DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_C);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
+            TimeRange.fromStartEnd(TIME_0830AM, TIME_0900AM, false),
+            TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void everyAttendeeIsConsideredAndOptionalIncluded() {
+    // Have each person have different events and an optional attendee who has a meeting exactly
+    // when the mandatory attendees have a free spot. This spot will be eliminated and return
+    // only the beggining of the day and the end o f the day.
+    //
+    // Events  :       |--A--|     |--B--|
+    // Optional:             |--C--|
```

Comment:
Optional: An interesting additional corner case here would be what would happen if between A and B's events, optional attendee C has an event, but there is still room for to schedule a meeting.

E.g. Are the intervals that work for A, B modified to match C's availability (better and I think what the code does) or just removed if they don't match C's availability.

Consider also some overlap between required and optional attendees.